+siibra/+core/Parcellation.m

```diff
@@ -1,32 +1,32 @@
 classdef Parcellation < handle
     properties
-        Id
-        Name
-        Atlas
-        Modality
-        Description
-        Graph
-        Spaces
+        id
+        name
+        atlas
+        modality
+        desciption
+        regionTree
+        spaces
     end
 
     methods
         function parcellation = Parcellation(parcellation_json, atlas)
-            parcellation.Id = strcat(parcellation_json.id.kg.kgSchema, '/', parcellation_json.id.kg.kgId);
-            parcellation.Name = parcellation_json.name;
-            parcellation.Atlas = atlas;
-            parcellation.Modality = parcellation_json.modality;
+            parcellation.id = strcat(parcellation_json.id.kg.kgSchema, '/', parcellation_json.id.kg.kgId);
+            parcellation.name = parcellation_json.name;
+            parcellation.atlas = atlas;
+            parcellation.modality = parcellation_json.modality;
             if ~ isempty(parcellation_json.infos)
-                parcellation.Description = parcellation_json.infos(1).description;
+                parcellation.desciption = parcellation_json.infos(1).description;
             end
 
             % link spaces from atlas
-            parcellation.Spaces = Space.empty;
+            parcellation.spaces = siibra.core.Space.empty;
             % retrieve available spaces from atlas
             for available_space_index = 1:numel(parcellation_json.availableSpaces)
                 % store handle to space object
-                for atlas_space_index = 1:numel(atlas.Spaces.Space)
-                    if isequal(atlas.Spaces.Space(atlas_space_index).ID, parcellation_json.availableSpaces(available_space_index).id)
-                        parcellation.Spaces(end +1) = atlas.Spaces.Space(atlas_space_index);
+                for atlas_space_index = 1:numel(atlas.spaces)
```

Comment:
Can consider iterating through the loop in reverse, e.g. `numel(atlas.spaces):-1:1`. 

That forces the JIT compiler to preallocate the array up front, avoiding the `end+1` dynamic resizing

Reply:
Okay, good to know. Would you say it is worth the mental overhead? Here we just store handles so no large objects have to be copied.